### PR TITLE
Vn/campaign summary scheduler

### PR DESCRIPTION
**Pull request by neal-bpm**
## Summary                                                           
                                                                       
  Adds automated posting of campaign delivery summaries to Slack after campaigns end.                                                       
                                                                       
  ## What This Does                                                    
                                                                       
  A new GenServer (`CampaignSummaryPoster`) runs every 15 minutes to find campaigns that ended 60-75 minutes ago and posts their delivery summaries to the program's configured Slack channel.                 
                                                                       
  ### Key Components                                                   
                                                                       
  **CampaignDeliverySummary**                                          
  (`lib/bike_brigade/delivery/campaign_delivery_summary.ex`)           
  - Aggregates delivery statistics: total/completed counts, per-rider breakdowns, unassigned deliveries                                    
  - Implements `Collectable` protocol for use with `Enum.into/2`       
                                                                       
  **CampaignSummaryPoster**                                            
  (`lib/bike_brigade/campaign_summary_poster.ex`)                      
  - Singleton GenServer running across the cluster                     
  - Checks every 15 minutes for recently ended campaigns               
  - 60-75 minute delay allows riders time to mark tasks complete       
  - Tracks posted summaries in database to prevent duplicates          
                                                                       
  **SlackCampaignSummary**                                             
  (`lib/bike_brigade/messaging/slack_campaign_summary.ex`)             
  - Schema for tracking which summaries have been sent                 
  - Unique constraint on `campaign_id` enables race-condition-free     inserts                                                              
                                                                       
  **Slack Block Kit Formatting**                                       
  (`lib/bike_brigade/slack_api/payload_builder.ex`)                    
  - Header with program name and campaign link                         
  - Date/time range (handles same-day vs multi-day)                    
  - Completed/total delivery counts                                    
  - Per-rider task lists (condensed view for campaigns with >15 tasks) 
  - Unassigned deliveries section when applicable                      
                                                                       
  ## Changes                                                           
                                                                       
  - Add `list_campaigns_ended_between/2` query in Delivery context     
  - Add CampaignSummaryPoster to application supervision tree          
  - Add `slack_campaign_summaries` table with unique constraint on   campaign_id                                                          
  - Handle missing Slack channel configuration gracefully with        notification                                                         
  - Sanitize campaign names and items for Slack formatting             
                                                                       
  ## Test Coverage                                                     
                                                                       
  - CampaignDeliverySummary aggregation and struct building            
  - PayloadBuilder Slack formatting, escaping, and ordering            
  - Delivery context time-window queries                               
  - Edge cases: empty campaigns, unassigned tasks, multi-day ranges 

## Checklist before requesting a review

- [X] I have performed a self-review of my code
- [X] If it is a core feature, I have added tests.
- [ ] Are there other PRs or Issues that I should link to here?
- [ ] Will this be part of a product update? If yes, please write one phrase
      about this update in the description above.

### CODE REVIEW

File: lib/bike_brigade/campaign_summary_poster.ex [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  end
+
+  defp do_post_summary(campaign, nil) do
+    Logger.warning("Skipping campaign #{campaign.id}: no Slack channel configured for program")
+    Slack.Operations.notify_campaign_error(campaign, "No Slack channel configured")
+  end
+
+  defp do_post_summary(campaign, channel_id) do
+    {_riders, tasks} = Delivery.campaign_riders_and_tasks(campaign)
+    summary = Enum.into(tasks, CampaignDeliverySummary.new(campaign))
```
Comment (mveytsman) [inline lib/bike_brigade/campaign_summary_poster.ex]: These should be encapsulated inside the CampaignDeliverySummary so you call something like

`CampaignDeliverySummary.create_for(campaign)`

**Review by mveytsman — CHANGES_REQUESTED** [review]
Some design feedback otherwise it's good!

**Review by mveytsman — APPROVED** [review]
🎉 looks great
